specs-go/v1/config.go

```diff
@@ -75,6 +75,7 @@ type History struct {
 }
 
 // Image is the JSON structure which describes some basic information about the image.
+// This provides the `application/vnd.oci.image.config.v1+json` mediatype when marshalled to JSON.
```

Comment:
> Now, you might argue that we should send the out-of-band media type with it. But why should that be a requirement?

That's exactly what I'll argue ;).  And unless you implement [completely generic peek-inside detection][1] (which I don't think anyone's arguing for), you're going to have to transmit *some* amount of media-type guidance along with your blob content.  I'm suggesting that guidance be the media type.

You seem to be suggesting that that guidance be “this blob is a JSON object which contains a self-describing `mediaType` field”.  Maybe you transmit that information because the tool-caller knows the tool can only handle such media types and therefore only feeds matching blobs into the tool.  That's how `oci-image-validate` works, and [I'm comfortable with that][2] from a keystroke-saving perspective.  However, I don't think we should pretend that this approach is completely free of out-of-band type guidance.

> What are you gaining by removing `mediaType`?

I'm not suggesting we remove `mediaType`, because some users (e.g. you with `oci-do-something`, or a number of people with `oci-image-validate`'s autodection) can't be bothered to pass media types around.  And I'm fine with that (typing out a long media type is not something I'd like to do repeatedly).

I'm just suggesting image-handling tools follow the spec's [SHOULD][3] and use descriptors to reference blob content, with peek-inside type detection being reserved for [signed-assertions][4].  And having acquired the media type from the referencing descriptor (or because we authored the blob ourselves), I see no need for image-handling tools to use peek-inside type detection.

Perhaps our difference here is that I see (almost) all tooling as being descriptor-based, while you see the tooling as being isolated-blob based.  Since I'm fine leaving existing `mediaType` entries in place, maybe we can just wait a year to see how that plays out and revisit this discussion then?

[1]: https://github.com/opencontainers/image-spec/pull/411#discussion_r86466669
[2]: https://github.com/opencontainers/image-spec/pull/411#issuecomment-257489701
[3]: https://github.com/opencontainers/image-spec/blame/v1.0.0-rc1/descriptor.md#L10
[4]: https://github.com/opencontainers/image-spec/pull/411#discussion_r86466669